### PR TITLE
Max: fix the subset name not changing accordingly after the variant name changes

### LINKED ISSUE #5902: Max: The context data not being updated correctly when the variant data has been changed.
Did one last test in ```max 2024``` and even tho it produced the ```Review``` without any issues, when tried to re publish it again, the Publish Instance looses its context?

@moonyuet can u confirm too??
image

_Originally posted by @LiborBatek in https://github.com/ynput/OpenPype/pull/5839#pullrequestreview-1726708115_
            

[cuID:[OP-7395](https://app.clickup.com/t/86bwdz27d)]

### PR DESCRIPTION
**Pull request by moonyuet**
## Changelog Description
Resolve #5902

## Additional info
N/A
## Testing notes:
1. Create multiple instances in the same family with OP in Max
2. If you create instance with the same subset names, it will error out.
3. Go to Publish Tab
4. Changes the Variant Name, Asset Name and Task Name
5. If user changes the variant name of which already been used as subset name, it will raise the error too.
6. Make sure you have different variant names for subset, and publish. It is supposed that it won't have wrong error on publishing.

### CODE REVIEW

File: openpype/hosts/max/api/plugin.py [hunk 1]
```diff
@@ -246,14 +248,25 @@ def collect_instances(self):
...
+            if subset:
+                if instance_node != subset:
+                    node = rt.getNodeByName(instance_node)
+                    new_subset_name = new_values["subset"]
+                    if rt.getNodeByName(new_subset_name):
+                        raise CreatorError(
+                            "The subset '{}' already exists.".format(
+                                new_subset_name))
+                    created_inst["instance_node"] = new_values["subset"]
+                    node.name = created_inst["instance_node"]
```
Comment (iLLiCiTiT) [inline openpype/hosts/max/api/plugin.py:265]:
```suggestion
            if subset and instance_node != subset:
                node = rt.getNodeByName(instance_node)
                new_subset_name = new_values["subset"]
                if rt.getNodeByName(new_subset_name):
                    raise CreatorError(
                        "The subset '{}' already exists.".format(
                            new_subset_name))
                instance_node = new_subset_name 
                created_inst["instance_node"] = instance_node 
                node.name = instance_node 
```

File: openpype/hosts/max/api/plugin.py [hunk 2]
```diff
@@ -246,14 +248,25 @@ def collect_instances(self):
...
+
             imprint(
-                instance_node,
-                new_values,
+                created_inst["instance_node"],
```
Comment (iLLiCiTiT) [inline openpype/hosts/max/api/plugin.py:268]: This is possible only with [this](https://github.com/ynput/OpenPype/pull/5911/files#r1392270231) suggestion
```suggestion
                instance_node,
```

File: openpype/hosts/max/api/lib.py [hunk 3]
```diff
@@ -42,6 +42,7 @@ def imprint(node_name: str, data: dict) -> bool:
             rt.SetUserProp(node, k, f"{JSON_PREFIX}{json.dumps(v)}")
         else:
             rt.SetUserProp(node, k, v)
+            print(k)
```
Comment (iLLiCiTiT) [inline openpype/hosts/max/api/lib.py:45]: Probably forgotten debug print?
```suggestion
```

**Comment by moonyuet** [comment]
@LiborBatek can you please test to see if it has fixed the issue you mentioned in https://github.com/ynput/OpenPype/pull/5839#pullrequestreview-1726708115? Thanks!

File: openpype/hosts/max/api/lib.py (hunk 3 above)
Comment (iLLiCiTiT) [reply openpype/hosts/max/api/lib.py:45]: It was resolved, so it's not forgotten?

Comment (moonyuet) [reply openpype/hosts/max/api/lib.py]: Sorry I forgot....It's already been removed in the updated commit

**Review by LiborBatek — APPROVED** [review]
All seems working well. It rises error msg when trying to name new ```instance``` with the same ```variant``` as already present one. This also being true when changing variant later on via Publisher. Again correctly rising the same error msg.

image


When properly renamed to unique ```variant``` names all publishing goes well and without any issues.

image

**Pull request merged**